chia/wallet/trade_manager.py

```diff
@@ -289,6 +289,96 @@ async def cancel_pending_offer_safely(
+                        selected_coins: Set[Coin] = await wallet.select_coins(
+                            uint64(fee_to_pay - coin.amount),
+                            exclude=[coin],
+                        )
+                        selected_coins.add(coin)
+                    else:
+                        selected_coins = {coin}
+                    tx: TransactionRecord = await wallet.generate_signed_transaction(
+                        uint64(sum([c.amount for c in selected_coins]) - fee_to_pay),
+                        new_ph,
+                        fee=fee_to_pay,
+                        coins=selected_coins,
+                        ignore_max_send_amount=True,
+                    )
+                    if tx is not None and tx.spend_bundle is not None:
+                        bundles.append(tx.spend_bundle)
+                        all_txs.append(dataclasses.replace(tx, spend_bundle=None))
+                else:
+                    # ATTENTION: new_wallets
+                    txs = await wallet.generate_signed_transaction(
+                        [coin.amount], [new_ph], fee=fee_to_pay, coins={coin}, ignore_max_send_amount=True
+                    )
+                    for tx in txs:
+                        if tx is not None and tx.spend_bundle is not None:
+                            bundles.append(tx.spend_bundle)
+                            all_txs.append(dataclasses.replace(tx, spend_bundle=None))
+                fee_to_pay = uint64(0)
+
+                cancellation_addition = Coin(coin.name(), new_ph, coin.amount)
+                all_txs.append(
+                    TransactionRecord(
+                        confirmed_at_height=uint32(0),
+                        created_at_time=uint64(int(time.time())),
+                        to_puzzle_hash=new_ph,
+                        amount=uint64(coin.amount),
+                        fee_amount=fee,
+                        confirmed=False,
+                        sent=uint32(10),
+                        spend_bundle=None,
+                        additions=[cancellation_addition],
+                        removals=[coin],
+                        wallet_id=wallet.id(),
+                        sent_to=[],
+                        trade_id=None,
+                        type=uint32(TransactionType.INCOMING_TX.value),
+                        name=cancellation_addition.name(),
+                        memos=[],
+                    )
+                )
+        # Aggregate spend bundles to the first tx
+        if len(all_txs) > 0:
+            all_txs[0] = dataclasses.replace(all_txs[0], spend_bundle=SpendBundle.aggregate(bundles))
+        if secure:
+            for tx in all_txs:
+                await self.wallet_state_manager.add_pending_transaction(
+                    tx_record=dataclasses.replace(tx, fee_amount=fee)
+                )
+        else:
+            self.wallet_state_manager.state_changed("offer_cancelled")
```

Comment:
Shouldn't this state be sent, regardless of `secure`?

Reply:
In the secure mode it should send a notification after the transaction sent, I assume. I just copy the logic from the single cancelling API. It only sends notification for secure = false.